### PR TITLE
Improve coverage

### PR DESCRIPTION
**Pull request by cpenarrieta**
The purpose of this PR is to improve the code coverage of the test cases. There is still some room to improve coverage for the `Slider` component but I think its a good start.
image

### CODE REVIEW

**Comment by Monte9** [comment]
@cpenarrieta Appreciate the PR! This looks great! 👌 

Want to bring up a couple things before this gets merged:
1. What makes `Slider.js` so hard to test? Also the test coverage image looks really clean. How did you generate it?
2. There seem to be a couple tests failing. Details here: https://travis-ci.org/react-native-training/react-native-elements/builds/220371540

This is a great start! Thanks for owning this.

**Comment by cpenarrieta** [comment]
The test coverage is generated by istanbul. once you run `npm run test:coverage` it creates a `coverage` folder and you can access the html report there.

`Slider.js` have lots of events and animations that are not to that straight forward to test. I'll need to spend more time on it.

I'll fix the build and let you know

**Comment by cpenarrieta** [comment]
@Monte9 build is passing now.

**Comment by Monte9** [comment]
Great. OK, so right now, `coverage` folder is not part of the commit as it's on `.gitignore`. What are your thoughts on removing it from `.gitignore`?

Here's an idea: we can create a `testing.md` file and have this image displayed on there along with other testing relevant information. This image could be replaced dynamically using Travis CI.

**Comment by Monte9** [comment]
Also @binoy14 & @bozzmob can review this PR when you guys get a chance?

**Comment by cpenarrieta** [comment]
If we include `coverage` folder in the build we'll have to make run that every PR executes `test:coverage`.
Another option is to use [codecov](https://codecov.io/). Here is an example https://github.com/facebook/jest/pull/3258

**Comment by cpenarrieta** [comment]
> Here's an idea: we can create a testing.md file and have this image displayed on there along with other testing relevant information. This image could be replaced dynamically using Travis CI.

I like this idea but I'm not sure how to do it

**Comment by binoy14** [comment]
Integrating codecov would be the ideal way to do it. @cpenarrieta do you have any experience with it?

**Comment by cpenarrieta** [comment]
I don't have experience setting it up but I can definitely find out.

**Comment by Monte9** [comment]
@cpenarrieta that would be great! Let us know what you learn and @binoy14 and I can work with you to get it up and running.

**Comment by binoy14** [comment]
> There seem to be a couple tests failing. Details here: https://travis-ci.org/react-native-training/react-native-elements/builds/220371540

Travis should fail on failing test. We need to fix this as failing tests will pass thru this. @cpenarrieta @monte9 any ideas?

**Comment by binoy14** [comment]
Take a look at this https://github.com/codecov/example-node/blob/master/readme.md#jest

It should be very straight forward to set up

**Comment by Monte9** [comment]
@binoy14 did a quick google search but didn't find why it would be doing that. I suspect it has something to do with the exit code that the unit tests are returning?

Will need to look into this further.

**Comment by binoy14** [comment]
@cpenarrieta Do you think you can cherry-pick this commit https://github.com/react-native-training/react-native-elements/pull/285/commits/fbddc7b4f1c4bdbc5982694aec5a742900dd6ca3 into this branch? I added `enzyme-to-json` to generate some friendly snapshots. If not don't worry about it

**Comment by cpenarrieta** [comment]
@binoy14 sure. I'll try doing that and let you know if I have some questions.

**Comment by cpenarrieta** [comment]
@binoy14 your commit is now merged.

File: src/avatar/__tests__/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import toJson from 'enzyme-to-json';
+import Avatar from '../Avatar';
+
+describe('Badge Component', () => {
```
Comment (binoy14) [inline src/avatar/__tests__/Avatar.js]: @cpenarrieta Small thing, the name of the test should be Avatar Component

File: src/slider/Slider.js [hunk 2]
```diff
@@ -337,7 +336,7 @@ export default class Slider extends Component {
     };
 
     var touchOverflowStyle = this.getTouchOverflowStyle();
-
+
```
Comment (binoy14) [inline src/slider/Slider.js]: Could you please remove the spacing changes.

**Comment by binoy14** [comment]
@cpenarrieta Just few small changes, besides that everything looks good. You are awesome 🥇

**Comment by cpenarrieta** [comment]
@binoy14 np. I'm happy to contribute!

**Comment by Monte9** [comment]
@cpenarrieta lot of opportunities to contribute on this repo and some really exciting things are coming up on our roadmap. Let me know if this is something you are interested in.

**Comment by cpenarrieta** [comment]
@Monte9 yes I am. Please keep me in the loop.

**Comment by cpenarrieta** [comment]
@Monte9 @binoy14 I configured codecov in one of my projects. I'll submit a PR with the changes you need to make in `.travis.yml`. You'll have to create an account on [https://codecov.io/](https://codecov.io/) and grant codecov permission for this project.

**Comment by cpenarrieta** [comment]
@Monte9 @binoy14 I merged the latest changes from parent branch. Let me know if this is ready to merge now.

**Comment by Monte9** [comment]
@binoy14 approves. And it looks ready to me. 
I'm going to merge it in now. Good job on this PR @cpenarrieta ! 👏

**Pull request merged**